File: ruby/1/09.rb
str = "I couldn't believe that I could actually understand what I was reading : the phenomenal power of the human mind ."

map = str.split(/\s/).map do |word|
  word.size <= 4 ? word : word[0] + word[1...-1].chars.shuffle.join + word[-1]
end

puts map.join(" ")

File: python/1/09.py
import random

def random_shuffle_word(word):
    if len(word) <= 4:
        return word
    else:
        inner = list(word[1:-1])
        random.shuffle(inner)
        return word[0] + ''.join(inner) + word[-1]

def random_shuffle(s):
    return ' '.join(map(random_shuffle_word, s.split()))

s = input()
print(random_shuffle(s))
